testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/util/FlowUtil.java

```diff
@@ -0,0 +1,257 @@
+        return this;
+    }
+
+    public FlowUtil addSubFlowExecution(Requirement requirement, Consumer<FlowUtil> flowInitializer) {
+        return addSubFlowExecution("sf" + rand.nextInt(), "basic-flow", requirement, flowInitializer);
+    }
+
+    public FlowUtil addSubFlowExecution(String alias, String providerId, Requirement requirement, Consumer<FlowUtil> flowInitializer) {
+        return addSubFlowExecution(alias, providerId, requirement, maxPriority + 10, flowInitializer);
+    }
+
+    public FlowUtil addSubFlowExecution(String alias, String providerId, Requirement requirement, int priority, Consumer<FlowUtil> flowInitializer) {
+        AuthenticationFlowModel flowModel = createFlowModel(alias, providerId, null, false, false);
+        return addSubFlowExecution(flowModel, requirement, priority, flowInitializer);
+    }
+
+    public static AuthenticationFlowModel createFlowModel(String alias, String providerId, String desc, boolean topLevel, boolean builtIn) {
+        AuthenticationFlowModel flowModel = new AuthenticationFlowModel();
+        flowModel.setId(UUID.randomUUID().toString());
+        flowModel.setAlias(alias);
+        flowModel.setDescription(desc);
+        flowModel.setProviderId(providerId);
+        flowModel.setTopLevel(topLevel);
+        flowModel.setBuiltIn(builtIn);
+        return flowModel;
+    }
+
+    public FlowUtil addSubFlowExecution(AuthenticationFlowModel flowModel, Requirement requirement, Consumer<FlowUtil> flowInitializer) {
+        return addSubFlowExecution(flowModel, requirement, maxPriority + 10, flowInitializer);
+    }
+
+    public FlowUtil addSubFlowExecution(AuthenticationFlowModel flowModel, Requirement requirement, int priority, Consumer<FlowUtil> flowInitializer) {
+        maxPriority = Math.max(maxPriority, priority);
+
+        flowModel = realm.addAuthenticationFlow(flowModel);
+
+        AuthenticationExecutionModel execution = new AuthenticationExecutionModel();
+        execution.setRequirement(requirement);
+        execution.setAuthenticatorFlow(true);
+        execution.setPriority(priority);
+        execution.setFlowId(flowModel.getId());
+        execution.setParentFlow(currentFlow.getId());
+        realm.addAuthenticatorExecution(execution);
+
+        if (flowInitializer != null) {
+            FlowUtil subflow = newFlowUtil(flowModel);
+            flowInitializer.accept(subflow);
+        }
+
+        return this;
+    }
+
+    private List<AuthenticationExecutionModel> getExecutions() {
+        if (executions == null) {
+            List<AuthenticationExecutionModel> execs = realm.getAuthenticationExecutions(currentFlow.getId());
+            if (execs == null) {
+                throw new FlowUtilException("Can't get executions of unknown flow " + currentFlow.getId());
+            }
+            executions = new ArrayList<>(execs);
+        }
+        return executions;
+    }
+
+    public FlowUtil removeExecution(int index) {
```

Comment:
Index is never used, instead index 0 is always removed

Reply:
You're right... I started working on it but had to switch on something else... I remove it for now.